sql/core/src/test/scala/org/apache/spark/sql/sources/InsertSuite.scala

```diff
@@ -1078,11 +1101,13 @@ class InsertSuite extends DataSourceTest with SharedSparkSession {
       }
     }
     // There is one trailing default value referenced implicitly by the INSERT INTO statement.
-    withTable("t") {
-      sql("create table t(i int, s bigint default 42, x bigint) using parquet")
-      assert(intercept[AnalysisException] {
-        sql("insert into t values(1)")
-      }.getMessage.contains("expected 3 columns but found"))
+    withSQLConf(SQLConf.USE_NULLS_FOR_MISSING_DEFAULT_COLUMN_VALUES.key -> "false") {
```

Comment:
Same as above, could we also add a version of this test with the conf set to "true" to show that the intended behavior works? (This should be the same as the test code in this case before this PR.)

Reply:
Daniel, I see several tests in test("INSERT INTO statements with tables with default columns: positive tests") already cover this scenario.

Reply:
OK, if there are existing tests that cover it when the `SQLConf.USE_NULLS_FOR_MISSING_DEFAULT_COLUMN_VALUES` is set to true, then no need to add more tests for those cases then.

Reply:
(Please feel free to resolve any other code-review comments where that is the case.)